### DISCUSSION TITLE
Indicate requirements that need to be met in order to remove an OAuth account

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#17799] [Feature Requests]
## Background

A community member recently wrote in and asked how they might unlink their Twitter account (OAuth) from their DEV account.

My explanation summed up was:

> For DEV, we require at least one form of authorization beyond email.
>
> The best option for you would likely be to set up a Forem Account via https://account.forem.com/ and connect it to your DEV account via https://dev.to/settings/account. Forem is the platform beyond DEV, and you can find other Forem communities to potentially join on https://discover.forem.com/. 
>
> After you've connected Forem to your DEV account, you can detach your Twitter account from your DEV account via https://dev.to/settings/account.

The community member wrote back and was grateful for direction, but they mentioned

> You probably got that suggested already, but if not, it would be clearer if there was a disabled 'Disconnect Twitter' button, with an explanation to connect another authorization method in order to be able to disconnect.

I agree that it would be quite helpful if the "Remove Twitter" button would appear but be greyed out with an explanation to connect another authorization method for the option to disconnect.

## Change Requested

Currently, if you only have 1 OAuth account connected to your DEV account, then you do not see the option to Remove OAuth Associations (see pick below) under the "Danger Zone"

![Screen Shot 2022-06-01 at 11 27 24 AM](https://user-images.githubusercontent.com/16007075/171441993-f735bb65-850a-4ce3-8418-fd6c283ea52b.png)

Ideally, this section should still be viewable even with just 1 OAuth account connected, but the button to "Remove Twitter" (or GitHub/Forem Account) should be greyed out with an explanation that 1 OAuth account must always be connected to your account.

## Outcome

Hopefully this would help DEV community members to solve their OAuth issues on their own because they would have an explanation in place in the UI as to why they can't remove their own OAuth and what they need to do to make that happen.

**Comment by maestromac** [comment]
thanks for this suggestion @michael-tharrington !

**Comment by heliocola** [comment]
Hi there @michael-tharrington and @maestromac : do you need a help with this?
I just went to my profile and I want to disconnect from GitHub, setup my own password, and have a place where I can put other social platforms I am currently using it.

I think this feature here as well as this one [Allow to add Mastodon profile URLs](https://github.com/forem/forem/discussions/18913) would be really valuable to the current state of the internet (and 🤞 to the near future)!
